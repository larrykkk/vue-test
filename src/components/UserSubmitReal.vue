<template>
  <div>
    <form @submit.prevent="handleSubmitAsync">
      <input v-model="username" data-username />
      <input type="submit" />
    </form>
<span>summer</span>
    <div class="message" v-if="submitted">
      <div>嗨, {{ username }},</div>
      <div>你最喜歡的食物是: {{ favorFood }}</div>
    </div>
  </div>
</template>

<script>
import axios from "axios";
export default {
  name: "FormSubmitter",

  data() {
    return {
      submitted: false,
      username: "",
      favorFood: "",
    };
  },

  methods: {
    handleSubmit() {
      this.submitted = true;
    },

    async handleSubmitAsync() {
      axios
        .post("/api/v1/submit", { username: this.username })
        .then((res) => {
          // console.log(res);
          this.submitted = true;
          this.username = res.data.payload.username;
          this.favorFood = res.data.payload.favorFood;
        })
        .catch((err) => {
          console.error(err);
        });
    },
  },
};
</script>
