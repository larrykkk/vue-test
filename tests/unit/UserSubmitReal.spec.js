import { mount } from "@vue/test-utils";
import UserSubmitReal from "@/components/UserSubmitReal.vue";
import flushPromises from "flush-promises";

jest.mock("axios", () => ({
  post: (_url, _data) =>
    Promise.resolve({
      data: {
        success: true,
        code: 200,
        payload: { username: _data.username, favorFood: "熱狗" },
      },
    }),
}));

describe("UserSubmitReal.vue", () => {
  it("當 submit 成功後將會顯示使用者最喜歡的食物", async () => {
    const wrapper = mount(UserSubmitReal);

    wrapper.find("[data-username]").setValue("larry");
    wrapper.find("form").trigger("submit.prevent");
    await flushPromises();

    console.log(wrapper.html());
    expect(wrapper.find(".message").text()).toBe(
      "嗨, larry, 你最喜歡的食物是: 熱狗"
    );
  });
});

describe("UserSubmitReal.vue", () => {
  it("當 submit 成功後將會顯示使用者最喜歡的食物", async () => {
    const wrapper = mount(UserSubmitReal);

    expect(wrapper.element).toMatchSnapshot();
  });
});
