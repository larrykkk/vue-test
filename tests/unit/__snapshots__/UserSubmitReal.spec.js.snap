// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSubmitReal.vue 當 submit 成功後將會顯示使用者最喜歡的食物 1`] = `
<div>
  <form>
    <input
      data-username=""
    />
     
    <input
      type="submit"
    />
  </form>
   
  <span>
    123
  </span>
   
  <!---->
</div>
`;
